packages/playwright/src/runner/watchMode.ts

```diff
@@ -147,32 +144,31 @@ export async function runWatchModeLoop(config: FullConfigInternal): Promise<Full
     const command = await readCommandPromise;
 
     if (command === 'changed') {
-      const dirtyTestFiles = fsWatcher.takeDirtyTestFiles();
-      // Resolve files that depend on the changed files.
-      await runChangedTests(config, failedTestIdCollector, dirtyTestFiles);
-      lastRun = { type: 'changed', dirtyTestFiles };
+      const testIds = [...dirtyTestIds];
+      dirtyTestIds.clear();
+      await runTests(options, testServerConnection, { testIds, title: 'files changed' });
+      lastRun = { type: 'changed', dirtyTestIds: testIds };
       continue;
     }
 
     if (command === 'run') {
       // All means reset filters.
-      await runTests(config, failedTestIdCollector);
+      await runTests(options, testServerConnection);
       lastRun = { type: 'regular' };
       continue;
     }
 
     if (command === 'project') {
-      const { projectNames } = await enquirer.prompt<{ projectNames: string[] }>({
+      const { selectedProjects } = await enquirer.prompt<{ selectedProjects: string[] }>({
         type: 'multiselect',
-        name: 'projectNames',
+        name: 'selectedProjects',
         message: 'Select projects',
-        choices: config.projects.map(p => ({ name: p.project.name })),
-      }).catch(() => ({ projectNames: null }));
-      if (!projectNames)
+        choices: teleSuiteUpdater.rootSuite!.suites.map(s => s.title),
```

Comment:
Won't this be a truncated list after we run `listTests()` with a project filter once? I think `teleSuiteUpdater.config.projects` is a better source.

Reply:
> I'm not sure what the right solution is. We could use the projects from our initial `listTests` call, but then we'd need to remove the filters from that. Would probably be fine.

This sounds like the easiest solution for now.

We can probably update `TeleEmitter` to include projects, but this needs to be done carefully to make sure merger is ok, in a follow up PR.

Reply:
oh no, I had an update to `listTests` typed but not committed before pushing 🤦 gonna work on that refactor next